### ISSUE TITLE
Custom entity Icons missing for Helpers in settings/entities table

### ISSUE THREAD

**Issue by Krispkiwi** [home-assistant/frontend#19896]
### Checklist

- [X] I have updated to the latest available Home Assistant version.
- [X] I have cleared the cache of my browser.
- [X] I have tried a different browser to see if it is related to my browser.
- [X] I have tried reproducing the issue in [safe mode](https://www.home-assistant.io/blog/2023/11/01/release-202311/#restarting-into-safe-mode) to rule out problems with unsupported custom resources.

### Describe the issue you are experiencing

There's an issue with Home Assistant's frontend not syncing entity icons on the entities tab. You can go through an re-icon most entities in the entities tab itself (despite them already being set on the devices tab) but not all, for example input booleans do not change over even when you attempt to manually fix it. 

https://imgur.com/a/pyTeF0X, also example here: https://imgur.com/a/I4KRGGR

I've tried multiple different instances and this one I'm taking the screenshot of is only hours old. Also asked in Lewis's discord and a few other people have noticed it.

### Describe the behavior you expected

Device icons in the entities tab should sync with the device icons set on the entity itself

### Steps to reproduce the issue

1. Add an entity such as a toggle helper or some other entities, I'm also seeing it with aqara products
2.  Go to the entities page, you'll see the entity icon doesn't match the one you see on the device

### What version of Home Assistant Core has the issue?

2024.2.4

### In which browser are you experiencing the issue with?

All of them

### Which operating system are you using to run this browser?

Windows 10

### State of relevant entities

```txt
It can be any type of entity but some are more prominent such as helper toggles which are on or off
```

**Comment by karwosts** [comment]
I agree something is not right here for helper entities.

The table seems to want to find an icon in the entity registry, but it seems for helpers when you set a custom icon, that doesn't go in the entity registry, but goes in some kind of config entry (and is still an attribute as part of the state object?)

**Comment by ildar170975** [comment]
Confirm this, all icons are missing (Chrome, iOS companion):

![image](https://github.com/home-assistant/frontend/assets/71872483/1469267f-9d3b-4d21-a3a2-c38f3c55967e)

The 1st row is for a helper made in UI.
It has a default icon.
Other rows - for helpers defined in yaml with custom icons.
All icons are nit displayed.

**Comment by ildar170975** [comment]
A related issue: for an `input_boolean` helper with a customized icon - a standard icon is shown:

![image](https://github.com/home-assistant/frontend/assets/71872483/a1c39f65-7775-4b51-a78d-fbad3d33dd63)

![image](https://github.com/home-assistant/frontend/assets/71872483/e7713a0e-26e1-43b0-9615-639641b3e679)

**Comment by ildar170975** [comment]
Seems to be fixed in 2024.4.1

**Comment by ildar170975** [comment]
> It resolved itself by restarting my HA Docker. (And thats probably what 'fixed' it on your update to 2024.4.1)

You cannot update HA w/o restarting a container, so yes, you can say "was fixed only after restarting a container" ))

**Comment by ildar170975** [comment]
Some browser cache related issues, may be...
